pkg/skaffold/build/acr/acr_build.go

```diff
@@ -0,0 +1,178 @@
+	if err != nil {
+		return "", errors.Wrap(err, "build source upload url")
+	}
+	blob := NewBlobStorage(*result.UploadURL)
+
+	err = docker.CreateDockerTarGzContext(blob.Buffer, artifact.Workspace, artifact.DockerArtifact)
+	if err != nil {
+		return "", errors.Wrap(err, "create context tar.gz")
+	}
+
+	err = blob.UploadFileToBlob()
+	if err != nil {
+		return "", errors.Wrap(err, "upload file to blob")
+	}
+
+	imageTag, err := tagger.GenerateFullyQualifiedImageName(artifact.Workspace, &tag.Options{
+		Digest:    util.RandomID(),
+		ImageName: artifact.ImageName,
+	})
+	if err != nil {
+		return "", errors.Wrap(err, "create fully qualified image name")
+	}
+
+	imageTag, err = getImageTagWithoutFQDN(imageTag)
+	if err != nil {
+		return "", errors.Wrap(err, "get azure image tag")
+	}
+
+	buildRequest := cr.DockerBuildRequest{
+		ImageNames:     &[]string{imageTag},
+		IsPushEnabled:  &[]bool{true}[0], //who invented bool pointers
+		SourceLocation: result.RelativePath,
+		Platform: &cr.PlatformProperties{
+			Variant:      cr.V8,
+			Os:           cr.Linux,
+			Architecture: cr.Amd64,
+		},
+		DockerFilePath: &artifact.DockerArtifact.DockerfilePath,
+		Type:           cr.TypeDockerBuildRequest,
+	}
+	future, err := client.ScheduleRun(ctx, b.ResourceGroup, b.ContainerRegistry, buildRequest)
+	if err != nil {
+		return "", errors.Wrap(err, "schedule build request")
+	}
+
+	run, err := future.Result(client)
+	if err != nil {
+		return "", errors.Wrap(err, "get run id")
+	}
+	runID := *run.RunID
+
+	runsClient := cr.NewRunsClient(b.Credentials.SubscriptionID)
+	runsClient.Authorizer = client.Authorizer
+	logURL, err := runsClient.GetLogSasURL(ctx, b.ResourceGroup, b.ContainerRegistry, runID)
+	if err != nil {
+		return "", errors.Wrap(err, "get log url")
+	}
+
+	err = streamBuildLogs(*logURL.LogLink, out)
+	if err != nil {
+		return "", errors.Wrap(err, "polling build status")
+	}
+
+	return imageTag, nil
+}
+
+func streamBuildLogs(logURL string, out io.Writer) error {
+	offset := int32(0)
+	for {
+		resp, err := http.Get(logURL)
+		if err != nil {
+			return err
+		}
+
+		if resp.StatusCode == http.StatusNotFound {
+			//if blob is not available yet, try again
+			time.Sleep(2 * time.Second)
+			continue
+		}
+
+		scanner := bufio.NewScanner(resp.Body)
+		line := int32(0)
+		for scanner.Scan() {
+			if line >= offset {
+				out.Write(scanner.Bytes())
+				out.Write([]byte("\n"))
+				offset++
+			}
+			line++
+		}
+		resp.Body.Close()
+
+		if offset > 0 {
+			switch resp.Header.Get(BuildStatusHeader) {
+			case "":
+				continue
+			case "internalerror":
+			case "failed":
+				return errors.New("run failed")
+			case "timedout":
+				return errors.New("run timed out")
+			case "canceled":
+				return errors.New("run was canceled")
+			default:
+				return nil
+			}
+		}
+
+		time.Sleep(2 * time.Second)
+	}
+}
+
+// ACR needs the image tag in the following format
+// <repository>:<tag>
+func getImageTagWithoutFQDN(imageTag string) (string, error) {
```

Comment:
Could you add a few test for this piece of code? Could you replace the manual parsing a an image name with something like `docker.ParseReference()` in `github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker` package?

Reply:
`docker.ParseReference()` only provides the `BaseName` consisting of `fqdn/repository`, but Azure only needs the repository name. I could extend the function to extract the repository name from the BaseName.